testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeDatabricksCloneTableCompatibility.java

```diff
@@ -0,0 +1,171 @@
+
+            onDelta().executeQuery(INSERT_INTO_DEFAULT + baseTable + " VALUES (1)");
+
+            // asserting there are only 1 rows when read from both spark and Trino
+            List<QueryAssert.Row> expectedRows = ImmutableList.of(row(1));
+            assertThat(onDelta().executeQuery(SELECT_FROM_DEFAULT + baseTable))
+                    .containsOnly(expectedRows);
+            assertThat(onTrino().executeQuery(SELECT_FROM_DELTA_DEFAULT + baseTable))
+                    .containsOnly(expectedRows);
+
+            onDelta().executeQuery(INSERT_INTO_DEFAULT + baseTable + " VALUES (2)");
+
+            // creating a shallow cloned table from version 1 of base Table
+            onDelta().executeQuery(CREATE_TABLE_DEFAULT + shallowClonedTableV1 + SHALLOW_CLONE_DEFAULT + baseTable + VERSION_AS_OF_1);
+
+            // asserting there are only 1 rows when read from
+            // spark as of VERSION 1 on base table and from TRINO on shallow cloned table
+            List<QueryAssert.Row> expectedRowsV1 = ImmutableList.of(row(1));
+            assertThat(onDelta().executeQuery(SELECT_FROM_DEFAULT + baseTable + VERSION_AS_OF_1))
+                    .containsOnly(expectedRowsV1);
+            assertThat(onTrino().executeQuery(SELECT_FROM_DELTA_DEFAULT + shallowClonedTableV1))
+                    .containsOnly(expectedRowsV1);
+
+            // creating a shallow cloned table from version 2 of base Table
+            onDelta().executeQuery(CREATE_TABLE_DEFAULT + shallowClonedTableV2 + SHALLOW_CLONE_DEFAULT + baseTable + VERSION_AS_OF_2);
+
+            // asserting the rows and the count to be 2 when the base table read from both Delta and Trino
+            // and also between base table version 2 and shallowClonedTableV2 table
+            List<QueryAssert.Row> expectedRowsV2 = ImmutableList.of(row(1), row(2));
+            assertThat(onDelta().executeQuery(SELECT_FROM_DEFAULT + baseTable))
+                    .containsOnly(expectedRowsV2);
+            assertThat(onTrino().executeQuery(SELECT_FROM_DELTA_DEFAULT + baseTable))
+                    .containsOnly(expectedRowsV2);
+            assertThat(onDelta().executeQuery(SELECT_FROM_DEFAULT + baseTable + VERSION_AS_OF_2))
+                    .containsOnly(expectedRowsV2);
+            assertThat(onTrino().executeQuery(SELECT_FROM_DELTA_DEFAULT + shallowClonedTableV2))
+                    .containsOnly(expectedRowsV2);
+        }
+        finally {
+            dropDeltaTableWithRetry(DEFAULT + baseTable);
+            dropDeltaTableWithRetry(DEFAULT + shallowClonedTableV1);
+            dropDeltaTableWithRetry(DEFAULT + shallowClonedTableV2);
+        }
+    }
+
+    @Test(groups = {DELTA_LAKE_DATABRICKS, DELTA_LAKE_OSS, DELTA_LAKE_EXCLUDE_73, DELTA_LAKE_EXCLUDE_91, PROFILE_SPECIFIC_TESTS})
+    @Flaky(issue = DATABRICKS_COMMUNICATION_FAILURE_ISSUE, match = DATABRICKS_COMMUNICATION_FAILURE_MATCH)
+    public void testShallowClonePartitionedTable()
```

Comment:
This test is very similar to `testShallowCloneTable`. Could you try introducing a helper method and call from `testShallowCloneTable` & `testShallowClonePartitionedTable`? You can refer to `TestDeltaLakeDatabricksCheckpointsCompatibility#testWriteStatsAsStructEnabled`.

Reply:
handled `Shallow clone/Deep clone * partitioned/non-partitioned` using `data-providers`